Feature: Firebase service account (#498)
codemagic-ci-cd/codemagic-docs: content/yaml-publishing/distribution.md

### Publishing an app to Firebase App Distribution

Codemagic enables you to automatically publish your iOS or Android app to [Firebase Console](https://console.firebase.google.com/).

For distributing an iOS application to Firebase App Distribution, your application must use a development, Ad Hoc or Enterprise distribution profile.

> To distribute your application Codemagic requires either **Firegase token** or service account with **Firebase App Distribution Admin** role. 

Reviewer: ```suggestion
To authenticate with Firebase, Codemagic requires either a **Firebase token** or a service account with **Firebase App Distribution Admin** role. 
```

To retrieve your Firebase token, follow the instructions in [Firebase documentation](https://firebase.google.com/docs/cli#cli-ci-systems).

> Note that using service account is a more secure option due the granular permission settings.

Reviewer: ```suggestion
Note that using a service account is a more secure option due to granular permission settings.
```

> Whether you use Firebase token or service account, add it to your environment variables in application or team settings. Reference it in `codemagic.yaml` as `$FIREBASE_TOKEN` or `$FIREBASE_SERVICE_ACCOUNT` respectively.

Reviewer: ```suggestion
Save the Firebase token or the contents of the service account JSON file to your environment variables in the application or team settings. Click **Secure** to encrypt the value. Then, reference it in `codemagic.yaml` as `$FIREBASE_TOKEN` or `$FIREBASE_SERVICE_ACCOUNT` respectively.
```

Example configuration for publishing Android and iOS artifacts to Firebase:

```yaml
publishing:
  firebase:
    firebase_token: $FIREBASE_TOKEN
    # OR: firebase_service_account: $FIREBASE_SERVICE_ACCOUNT
    android:

>       app_id: x:xxxxxxxxxxxx:android:xxxxxxxxxxxxxxxxxxxxxx # Add your Android app id retrieved from Firebase console

Reviewer: let's add iOS app ID example too
